File: BTree.py
class BTreeNode:
    def __init__(self, t, is_leaf=True):
        self.t = t  # Minimum degree (defines the range for the number of keys)
        self.is_leaf = is_leaf
        self.keys = []  # List of keys
        self.children = []  # List of children pointers

class BTree:
    def __init__(self, t):
        self.root = BTreeNode(t)
        self.t = t

    def insert(self, key, location):
        # Insert logic for B-tree (to be implemented based on B-tree properties)
        pass

    def search(self, key, node=None):
        if node is None:
            node = self.root
        i = 0
        while i < len(node.keys) and key > node.keys[i][0]:
            i += 1
        if i < len(node.keys) and key == node.keys[i][0]:
            return node.keys[i][1]  # Return location associated with file_name
        if node.is_leaf:
            return None
        return self.search(key, node.children[i])

# Example usage:
btree = BTree(3)  # Minimum degree of 3
btree.insert("file1.txt", "/path/to/file1")
print(btree.search("file1.txt"))  # Should print "/path/to/file1"

File: BTree.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class BTreeNode {
public:
    vector<pair<string, string>> keys;  // (file_name, location)
    vector<BTreeNode*> children;
    bool isLeaf;
    int t;  // Minimum degree

    BTreeNode(int _t, bool _isLeaf) : t(_t), isLeaf(_isLeaf) {}

    void insertNonFull(const string& key, const string& location);
    BTreeNode* search(const string& key);

    // Additional functions for splitting children and inserting in the tree would be added
};

class BTree {
public:
    BTreeNode* root;
    int t;  // Minimum degree

    BTree(int _t) : t(_t) {
        root = new BTreeNode(_t, true);
    }

    void insert(const string& key, const string& location);
    BTreeNode* search(const string& key) {
        return (root == nullptr) ? nullptr : root->search(key);
    }
};

// Implementations of insertion and search functions to be added
